app2.py

```diff
@@ -0,0 +1,264 @@
+        args = parser.parse_args()
+        # args, unknown = parser.parse_known_args()
+        self.username = args.username
+        self.password = args.password
+
+    #----- Utility methods -----
+    def _set_session(self, token: str) -> None:
+        self.headers = {'Authorization': f'Bearer {token}'}
+        self.session.headers.update(self.headers)
+
+    def _set_url(self, endpoint: str) -> str:
+        url = ''.join([self.url, endpoint])
+        return url
+
+    def _get_token_for_group(self, group: str) -> str:
+        # curl -u $user:$password -XPOST "https://artitestserver.jfrog.io/artifactory/api/security/token" -d "username=$user" -d "scope=member-of-groups:readers" > .tok3
+        
+        endpoint = 'security/token'
+        # url = ''.join([self.url, endpoint])
+        url = self._set_url(endpoint)
+        data = {'username':self.username, 'password':self.password, 'scope=member-of-groups':group}
+        r = requests.post(url, auth=(self.username, self.password), data=data)
+        # r = self.session.post(url, auth=(self.username, self.password), data=self.data)
+        return r.json()['access_token']
+
+    #----- Menu options implementation -----
+    # 1
+    def _system_ping(self) -> requests.Response:
+        # curl -H "Authorization: Bearer $Access_Token" https://artitestserver.jfrog.io/artifactory/api/system/ping
+        endpoint = 'system/ping'
+        url = self._set_url(endpoint)        
+        r = self.session.get(url, headers=self.session.headers)
+        return r
+
+    # 2
+    def _system_version(self) -> str:
+        # curl -H "Authorization: Bearer $Access_Token" https://artitestserver.jfrog.io/artifactory/api/system/ping
+        endpoint = 'system/version'
+        url = self._set_url(endpoint)    
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()['version']
+
+    # 3
+    def _create_user(self) -> requests.status_codes:
+        # Notes: Requires Artifactory Pro
+        # Usage: PUT /api/security/users/{userName}
+        username = input("Enter username: ")
+        email = input("Enter email: ")
+        password = input("Enter password: ")
+        if (username or email or password) is (None or ""):
+            raise ValueError("Missing one or more values (username, email, password)")
+            
+        endpoint = f'security/users/{username}'
+        url = self._set_url(endpoint)
+        # email and password are mandatory, name is optional
+        json = {'name': f'{username}',
+                'email' : f'{email}',
+                'password': f'{password}'
+                }
+
+        r = self.session.put(url, headers=self.session.headers, json=json)
+        return r.status_code
+
+    # 4 
+    def _delete_user(self) -> str:
+        # Notes: Requires Artifactory Pro
+        # Usage: DELETE /api/security/users/{userName}
+        username = input("Enter username to delete: ")
+        if username == "" or username is None:
```

Comment:
I think it is better to check this first `username is None` because maybe some python version will raise an error when checking if `None` equals something